rosetta-testing/src/test/java/com/regnosys/rosetta/types/SubtypeRelationTest.xtend

```diff
@@ -28,6 +36,166 @@ class SubtypeRelationTest {
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB number (1..1)
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val result = fieldA.join(fieldB)
+		assertTrue(result.RType instanceof RNumberType)
+		assertFalse(result.hasMeta)
+	}
+	
+	@Test
+	def testJoinOnSameBaseTypeWithMetaIsCorrect() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata scheme]
+				[metadata address]
+		'''.parseAttribute.RTypeOfSymbol
+				
+		val result = fieldA.join(fieldB)
+		assertTrue(result.RType instanceof RStringType)
+		assertEquals(#[new RMetaAttribute("scheme", UNCONSTRAINED_STRING)], result.getMetaAttributes)
+	}
+	
+	@Test
+	def testStringWithSchemeAndReferenceHasNoSubtypeRelationWithStringWithAddressAndLocation() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata address]
+				[metadata location]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertFalse(fieldA.isSubtypeOf(fieldB))
+		assertFalse(fieldB.isSubtypeOf(fieldA))
+	}
+	
+	@Test
+	def testStringWithShemeIsSubtypeOfStringWithScheme() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertTrue(fieldA.isSubtypeOf(fieldB))
+		assertTrue(fieldB.isSubtypeOf(fieldA))
+	}
+	
+	@Test
+	def testStringWithSchemeIsNotSubtypeOfStringWithSchemeAndReference() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertFalse(fieldB.isSubtypeOf(fieldA))
+	}
+	
+	@Test
+	def testStringWithSchemeAndReferenceIsSubTypeOfStringWithScheme() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertTrue(fieldA.isSubtypeOf(fieldB))
+	}
+	
+	@Test
+	def testStringIsNotSubTypeOfStringWithScheme() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertFalse(fieldB.isSubtypeOf(fieldA))
+	}
+	
+	@Test
+	def testStringWithSchemeIsSubTypeOfString() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+		'''.parseAttribute.RTypeOfSymbol
+		
+		assertTrue(fieldA.isSubtypeOf(fieldB))
+	}
```

Comment:
Last four tests seem to be testing the same code paths. I wonder if this is just not adding unnecessary maintenance.